src/quo2/components/code/snippet.cljs

```diff
@@ -0,0 +1,157 @@
+                                                            :locations [0.75 1]
+                                                            :style {:flex 1}}])}]
+                                 [:<>])]
+
+        [rn/view {:style {:background-color background-color
+                          :border-color border-color
+                          :border-width 1
+                          :border-radius 8
+                          :overflow :hidden
+                          :padding 8
+                          ;; Hide on intial render to avoid flicker when mask-wrapper is shown.
+                          :opacity (if @text-height 1 0)}}
+         ;; Line number container
+         [rn/view {:style {:position :absolute
+                           :bottom 0
+                           :top 0
+                           :left 0
+                           :width (+ line-number-width 8 8)
+                           :background-color background-color-left
+                           :border-right-color border-color
+                           :border-right-width 1}}]
+         (conj maybe-mask-wrapper [rn/view {:max-height max-text-height}
+                                   [rn/view {:on-layout (fn [evt]
+                                                          (let [height (oops/oget evt "nativeEvent.layout.height")]
+                                                            (reset! text-height height)))}
+                                    [code-block {:rows (take max-rows rows)
+                                                 :line-number-width line-number-width}]]])
+
+         ;; Copy button
+         [rn/view {:style {:position :absolute
+                           :bottom 8
+                           :right 8}}
+          [button/button {:icon true
+                          :type :grey
+                          :size 24
+                          :on-press on-copy-press}
+           :main-icons/copy]]]))))
+
+(defn- wrap-renderer-fn [f {:keys [max-lines on-copy-press]}]
+  (fn [^js props]
+    (reagent/as-element [:f> f {:rows (.-rows props)
+                                :max-lines max-lines
+                                :on-copy-press on-copy-press}])))
+
+(defn snippet [{:keys [language max-lines on-copy-press]} children]
+  [:> Highlighter {:language language
+                   :renderer (wrap-renderer-fn
+                              native-renderer {:max-lines max-lines
+                                               :on-copy-press #(when on-copy-press
+                                                                 (on-copy-press children))})
+                   ;; Default props to adapt Highlighter for react-native.
+                   :CodeTag react-native/View
+                   :PreTag react-native/View
+                   :style #js {}
+                   :customStyle #js {:backgroundColor nil}
+                   :showLineNumbers false}
```

Comment:
Can we make this kebab cased ?

Reply:
Updated all except `CodeTag` and `PreTag`, which are unfortunately not transformed by reagent. Thanks for noticing. :)